spec/requests/onboarding/telegram_spec.rb

```diff
@@ -1,170 +1,111 @@
-      it { is_expected.not_to raise_exception }
+    context 'without data processing consent' do
+      let(:data_processing_consent) { false }
 
-      it 'is successful' do
+      it 'displays validation errors' do
         subject.call
-        expect(response).to be_successful
+        parsed = Capybara::Node::Simple.new(response.body)
+        fields = parsed.all('.Field')
+        data_processing_consent_field = fields.find { |f| f.has_text? 'Allgemeine Nutzungsbedingungen' }
+        expect(data_processing_consent_field).to have_text('muss akzeptiert werden')
       end
 
-      it { is_expected.to change(Contributor, :count).by(1) }
-
-      it 'creates contributor' do
-        subject.call
-        expect(Contributor.first).to have_attributes(
-          first_name: 'Matthew',
-          last_name: 'Rider',
-          telegram_id: 123,
-          username: 'matthew_rider',
-          jwt: jwt
-        )
+      it 'does not create new contributor' do
+        expect { subject.call }.not_to change(Contributor, :count)
       end
+    end
 
-      it 'attaches telegram profile picture' do
-        subject.call
-        expect(Contributor.first.avatar).to be_attached
-      end
+    describe 'with unsigned jwt' do
+      let(:jwt) { 'INCORRECT_TOKEN' }
 
-      it 'invalidates the jwt' do
-        expect { subject.call }.to change(JsonWebToken, :count).by(1)
+      it 'renders unauthorized page' do
+        subject.call
 
-        json_web_token = JsonWebToken.where(invalidated_jwt: jwt)
-        expect(json_web_token).to exist
+        expect(response).not_to be_successful
       end
 
-      context 'sets an encrypted cookie' do
-        let(:cookie_jar) { ActionDispatch::Cookies::CookieJar.build(request, cookies.to_hash) }
-
-        it 'with telegram_id for use in next step' do
-          subject.call
-          expect(cookie_jar.encrypted['telegram_id']).to eq(123)
-        end
+      it 'does not create new contributor' do
+        expect { subject.call }.not_to change(Contributor, :count)
       end
     end
 
-    describe 'given an existing telegram id' do
-      let(:attrs) do
-        {
-          first_name: 'Tom',
-          last_name: 'Jones',
-          telegram_id: 123
-        }
-      end
-      let!(:contributor) { create(:contributor, **attrs) }
-
-      it 'redirects to success page' do
-        subject.call
-        expect(response).to redirect_to onboarding_success_path
-      end
-
-      it 'invalidates the jwt' do
-        expect { subject.call }.to change(JsonWebToken, :count).by(1)
+    describe 'with invalidated jwt' do
+      let!(:invalidated_jwt) { create(:json_web_token, invalidated_jwt: 'INVALID_JWT') }
+      let(:jwt) { 'INVALID_JWT' }
 
-        json_web_token = JsonWebToken.where(invalidated_jwt: jwt)
-        expect(json_web_token).to exist
+      describe 'http status' do
+        subject { super().call && response }
+        it { is_expected.to have_http_status(:unauthorized) }
```

Comment:
I’m not sure I understand this. Why do we need `super().call`?

Reply:
`have_http_status` is defined on `response`. In order to write `is_expected.to have_http_status(...)` I first call `super()` for the actual request and make `response` the new return value.